wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +37,380 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
+             fundingInputs = utxos.map(_._1),
+             remoteFundingInputs = Vector.empty,
+             feeRate = feeRate,
+             changeSPK = changeSPK,
+             remoteChangeSPK = P2WSHWitnessSPKV0(EmptyScriptPubKey),
+             cetSigs = CETSignatures(PartialSignature.dummyPartialSig,
+                                     PartialSignature.dummyPartialSig,
+                                     PartialSignature.dummyPartialSig),
+             fundingSigs =
+               FundingSignatures(Vector(PartialSignature.dummyPartialSig)),
+             oracleSig = DummySchnorrDigitalSignature,
+             network = network
+           )),
+       utxos)
+    }
+
+    dlcDbFs.flatMap {
+      case (dlcDbF, utxos) =>
+        dlcDbF.map { dlcDb =>
+          logger.debug("DLC Offer added to database")
+          DLCOffer(
+            dlcDb.contractInfo,
+            oracleInfo,
+            keyManager.getRootXPub, // fixme this will need to be changed according to KeyDerivation.md
+            amount.satoshis,
+            utxos,
+            Bech32Address(dlcDb.changeSPK, dlcDb.network),
+            feeRate,
+            dlcDb.timeouts
+          )
+        }
+    }
+  }
+
+  override def acceptDLCOffer(
+      offer: DLCOffer,
+      amount: Bitcoins): Future[DLCAccept] = {
+    logger.debug("Calculating relevant wallet data for DLC Accept")
+
+    val eventId =
+      DLCMessage.calcEventId(offer.oracleInfo,
+                             offer.contractInfo,
+                             offer.timeouts)
+
+    logger.debug("Checking if Accept has already been made")
+    for {
+      account <- getDefaultAccountForType(AddressType.SegWit)
+      dlcDbOpt <- dlcDAO.findByEventId(eventId)
+      accept <- dlcDbOpt match {
+        case Some(dlcDb) =>
+          logger.debug("DLC Accept has already been made, returning Accept")
+          listUtxos(dlcDb.fundingInputs).flatMap { spendingInfoDbs =>
+            val spendingInfos =
+              spendingInfoDbs.flatMap(_.toUTXOSpendingInfo(
+                account,
+                keyManager,
+                dlcDb.network).toSingles)
+            val client = BinaryOutcomeDLCClient.fromOffer(
+              offer,
+              keyManager.rootExtPrivKey, // todo change to a ExtSign.deriveAndSignFuture // fixme this will need to be changed according to KeyDerivation.md
+              spendingInfos,
+              amount,
+              dlcDb.changeSPK.asInstanceOf[P2WPKHWitnessSPKV0]
+            )
+            client.createCETSigs.map { cetSigs =>
+              DLCAccept(
+                amount.satoshis,
+                keyManager.getRootXPub,
+                spendingInfos.map(info => (info.outPoint, info.output)),
+                Bech32Address(dlcDb.changeSPK, dlcDb.network),
+                cetSigs,
+                eventId
+              )
+            }
+          }
+        case None =>
+          logger.debug("DLC Offer not found, creating a new one")
```

Comment:
This log seems weird, but i'm assuming this is going to change when we split out the tables.